request.go

```diff
@@ -1926,6 +1926,62 @@ func (r *Request) WithMultipart() *Request {
 	return r
 }
 
+//	req := NewRequestC(config, "GET", "/path")
+//	req.WithClient(&http.Client{
+//	  Transport: &http.Transport{
+//		DisableCompression: true,
+//	  },
+//	})
+
+// WithReporter sets reporter.
+//
+// The new reporter overwrites AssertionHandler.Reporter. It will be used to
+// report formatted fatal failure messages
+//
+// Example:
+//
+//	req := NewRequestC(config, "GET", "http://example.com/path")
+//	reporter := httpextect.ReporterFunc(
+//		func(message string, args ...interface{}) {
+//			// reporter code here
+//		}),
+//	req.WithReporter(reporter)
+func (r *Request) WithReporter(reporter Reporter) *Request {
+	return r.WithAssertionHandler(&DefaultAssertionHandler{
+		Reporter:  reporter,
+		Formatter: r.config.Formatter,
+	})
+}
```

Comment:
We shouldn't call WithAssertionHandler from WithReporter, because they should use different names for r.chain.enter. I think we can just copy-paste implementation, since it's small.